server/middleware.go

```diff
@@ -6,6 +6,7 @@ import (
 	"strings"
 
 	"github.com/gin-gonic/gin"
+	"github.com/mikeydub/go-gallery/copy"
 	"github.com/mikeydub/go-gallery/runtime"
 )
 
```

Comment:
For getting UserId from context we should be able to do this as a sort of context extension function.
```suggestion
func (c *gin.Context) getUserId() persist.DbId {
val, _ := c.Get(userIdContextKey)
return val.(persist.DbId)
}
```

Reply:
this would be awesome, but i'm not sure how to address this- as far as i know, you can only define methods on a type defined in that same package. 

to implement your strategy we'd probably need to create a new type then pass it all the way down right? either way, i'll merge this for now and maybe you can take a look later!